topics/basic.md

```diff
@@ -1,78 +1,110 @@
+   ```
+   {style="block"}
 
-You can use several strategies for validating:
+1. A server [validates](#configure-provider) credentials sent by a client and responds with the requested content.
 
-### Manual credential validation
 
-Since there is the `validate` callback for authentication, you can just put your code there.
-So you can do things like checking the password against a constant, authenticating using a database
-or composing several validation mechanisms.
+## Install basic authentication {id="install"}
+To install the `basic` authentication provider, call [basic](https://api.ktor.io/ktor-features/ktor-auth/ktor-auth/io.ktor.auth/basic.html) function inside the `install` block:
 
 ```kotlin
+install(Authentication) {
+    basic {
+        // Configure basic authentication
+    }
+}
 ```
-{src="snippets/auth-basic/src/main/kotlin/com/example/Application.kt" lines="9-18"}
 
-Remember that both the `name` and the `password` from the credentials are arbitrary values.
-Remember to escape and/or validate them when accessing with those values to the file system, a database,
-when storing them, or generating HTML with its content, etc.
-{ .security.note }
+You can optionally specify a [provider name](authentication.md#provider-name) that can be used to [authenticate a specified route](#authenticate-route).
+
+## Configure basic authentication {id="configure"}
 
-### Validating using UserHashedTableAuth
+To get a general idea on how to configure different authentication providers in Ktor, see [](authentication.md#configure). In this section, we'll see on configuration specifics of the `basic` authentication provider. 
 
-There is a class that handles hashed passwords in-memory to authenticate `UserPasswordCredential`.
-You can populate it from constants in code or from another source. You can use predefined digest functions
-or your own.
+### Step 1: Configure a basic provider {id="configure-provider"}
 
-*Instantiating:*
+The `basic` authentication provider exposes its settings via the [BasicAuthenticationProvider.Configuration](https://api.ktor.io/ktor-features/ktor-auth/ktor-auth/io.ktor.auth/-basic-authentication-provider/-configuration/index.html) class. In the example below, the following settings are specified:
+* The `realm` property sets the realm to be passed in `WWW-Authenticate` header.
+* The `validate` function validates a user name and password.
 
 ```kotlin
 ```
-{src="snippets/auth-basic-hash-table/src/main/kotlin/com/example/Application.kt" lines="9-16"}
+{src="snippets/auth-basic/src/main/kotlin/com/example/Application.kt" lines="9-20"}
+   
+The `validate` function checks `UserPasswordCredential` and returns a `UserIdPrincipal` in a case of successful authentication or `null` if authentication fails. 
+> You can also use [UserHashedTableAuth](#validate-user-hash) to validate users stored in an in-memory table that keeps user names and password hashes.
 
+### Step 2: Protect specific routes {id="authenticate-route"}
```

Comment:
In all instance where you say Protect specific routes, let's use the same as the general one that talks scopes and authorization.